Framework/Algorithms/src/SmoothNeighbours.cpp

```diff
@@ -414,41 +414,31 @@ void SmoothNeighbours::findNeighboursUbiqutious() {
         neighbour.second /= totalWeight;
 
     // Save the list of neighbours for this output workspace index.
-    m_neighbours[outWI] = neighbours;
-    outWI++;
+    m_neighbours[m_outWI] = neighbours;
+    m_outWI++;
 
     m_progress->report("Finding Neighbours");
   } // each workspace index
-
-  delete[] used;
```

Comment:
Modern C++ practices encourage users to avoid new/delete in favour of classes that manage the memory. In the first commit I switch `used` to a vector so it will do the deletion itself and it can't be missed if an exception is thrown.